test/run_test.sh

```diff
@@ -91,6 +91,20 @@ if [[ "$OSTYPE" != "msys" ]]; then
       rm -rf "$TEMP_DIR"
     }
 
+    gloo_multigpu_set_up() {
+      # When we have 2 GPUs, there're not enough GPUs for multigpu tests to run
+      # for 3 processes. Thus we spawn 2 processes and skip all dist.new_group
+      # tests. When we have 0 or 1 GPU, multigpu tests are skipped anyways.
+      num_gpu=$($PYCMD -c \
+	      $'import torch;print(torch.cuda.device_count() if torch.cuda.is_available() else 0)')
+      min_gpu_for_multigpu=2
+      if [ "$num_gpu" -eq "$min_gpu_for_multigpu" ]; then
+	export WORLD_SIZE=2
+      else
+	export WORLD_SIZE=3
```

Comment:
There's no cleanup, so this will affect all tests after gloo as well (mpi and nccl)

Reply:
This comment still hasn't been fixed